src/dataflow/render/reduce.rs

```diff
@@ -483,11 +483,15 @@ where
     collection
         .map(move |((key, hash), row)| ((key, hash % modulus), row))
         .reduce_named("ReduceHierarchical", {
-            move |_key, source, target| {
+            move |key, source, target| {
                 // Should negative accumulations reach us, we should loudly complain.
                 if source.iter().any(|(_val, cnt)| cnt <= &0) {
-                    // TODO(frank): Consider reporting the actual offending data.
-                    log::error!("Negative accumulation in ReduceHierarchical");
+                    for (val, cnt) in source.iter() {
+                        if cnt <= &0 {
+                            // XXX: This reports user data, which we perhaps should not do!
```

Comment:
the wording here could be taken the wrong way IMO, like "YOLO, we haven't thought about privacy!"

Maybe something like "// TODO - make printing of user data configurable" would be better? Use your judgment I guess

Reply:
I copy/pasted the other place in the file where we do this. I wanted something more scary sounding than "TODO" and "XXX" was what was recommended. Up for any language, but I want it to be as unmissable as possible.

Reply:
we could potentially have a string that we log in the output, to make it easier for customers to ensure that they know what they're sending us, e.g:

```rust
log::error!("[customer-data] Non-positive accumulation in ReduceHierarchical: key: {:?}\tvalue: {:?}\tcount: {:?}", key, val, cnt);
```